### PR TITLE
Change IDP storing structure in choreo connect

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/discovery/xds/server.go [hunk 1]
```diff
@@ -113,8 +112,8 @@ var (
 	enforcerThrottleData             *throttle.ThrottleData
 
 	// KeyManagerList to store data
-	KeyManagerList = make([]eventhubTypes.KeyManager, 0)
-	isReady        = false
+	// KeyManagerList = make([]eventhubTypes.KeyManager, 0)
```
Comment (VirajSalaka) [inline adapter/internal/discovery/xds/server.go:115]: Delete this

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/keymgt/KeyManagerHolder.java [hunk 2]
```diff
@@ -152,24 +150,57 @@ public void addKMTokenIssuers(String keyManagerName, Map<String, Object> configu
                         }
                     }
                 }
-                kmIssuerMap.put(tokenIssuerDto.getIssuer(), tokenIssuerDto);
+                if (!kmIssuerMap.containsKey(organization)) {
+                    kmIssuerMap.put(organization, new HashMap<>());
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/keymgt/KeyManagerHolder.java:154]: ConcurrentHashMap to be safe

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java [hunk 3]
```diff
@@ -185,15 +186,25 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
...
                     // Check if the token has access to the gateway configured environment.
                     checkTokenEnv(claims, requestContext.getMatchedAPI().getEnvironmentName());
                     // Validate subscriptions
                     APIKeyValidationInfoDTO apiKeyValidationInfoDTO = new APIKeyValidationInfoDTO();
-                    EnforcerConfig configuration = ConfigHolder.getInstance().getConfig();
-                    ExtendedTokenIssuerDto issuerDto = configuration.getIssuersMap().get(validationInfo.getIssuer());
+                    ExtendedTokenIssuerDto issuerDto = KeyManagerHolder.getInstance()
+                            .getTokenIssuerDTO(requestContext.getMatchedAPI().getOrganizationId(),
+                                    validationInfo.getIssuer());
+                    isAllowedEnvironmentForIDP(requestContext.getMatchedAPI().getEnvironmentName(),
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java:200]: Change order

**Review by malinthaprasan — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20230816.21

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230816.21

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20230816.21

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230816.21

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230816.23

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230816.23

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230816.23

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20230816.23

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230817.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230817.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230817.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230817.3

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : prod-deployment-v2 : 20230817.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230817.3

**Pull request merged**